### PR TITLE
Adding videos to chapters

### PR DESCRIPTION
**Pull request by leios**
This PR adds a video to each chapter with a video associated with it. I understand if people are against this.

### CODE REVIEW

**Comment by jiegillet** [comment]
I think it's a great idea. Synergy!

**Comment by leios** [comment]
Cool. I think most people are generally in favour of the idea, so I'll force merge it later (unless someone wants to approve it first)

File: contents/euclidean_algorithm/euclidean_algorithm.md [hunk 1]
```diff
@@ -142,6 +142,15 @@ Here, we set `b` to be the remainder of `a%b` and `a` to be whatever `b` was las
...
 The Euclidean Algorithm is truly fundamental to many other algorithms throughout the history of computer science and will definitely be used again later. At least to me, it's amazing how such an ancient algorithm can still have modern use and appeal. That said, there are still other algorithms out there that can find the greatest common divisor of two numbers that are arguably better in certain cases than the Euclidean algorithm, but the fact that we are discussing Euclid two millennia after his death shows how timeless and universal mathematics truly is. I think that's pretty cool.
 
+## Video Explanation
+
+Here's a video on the Euclidean algorithm:
+
+<div style="text-align:center">
+<iframe width="560" height="315" src="https://www.youtube.com/embed/h86RzlyHfUE" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+</div>
+
```
Comment (julianschacher) [inline contents/euclidean_algorithm/euclidean_algorithm.md]: A line too much.
```suggestion
```

File: contents/forward_euler_method/forward_euler_method.md [hunk 2]
```diff
@@ -91,6 +91,15 @@ Verlet integration has a distinct advantage over the forward Euler method in bot
...
 That said, variations of this method *are* certainly used (for example Crank-Nicolson and [Runge-Kutta](../runge_kutta_methods/runge_kutta_methods.md), so the time spent reading this chapter is not a total waste!
 
+## Video Explanation
+
+Here is a video describing the forward Euler method:
+
+<div style="text-align:center">
+<iframe width="560" height="315" src="https://www.youtube.com/embed/wG7h8g6VLBo" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+</div>
+
```
Comment (julianschacher) [inline contents/forward_euler_method/forward_euler_method.md]: A line too much.
```suggestion
```

File: contents/gaussian_elimination/gaussian_elimination.md [hunk 3]
```diff
@@ -529,6 +529,15 @@ How? Well, we can use an algorithm known as the _Tri-Diagonal Matrix Algorithm_
...
 There are also plenty of other solvers that do similar things that we will get to in due time.
 
+## Video Explanation
+
+Here's a video describing Gaussian elimination:
+
+<div style="text-align:center">
+<iframe width="560" height="315" src="https://www.youtube.com/embed/2tlwSqblrvU" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+</div>
+
```
Comment (julianschacher) [inline contents/gaussian_elimination/gaussian_elimination.md]: A line too much.
```suggestion
```

File: contents/monte_carlo_integration/monte_carlo_integration.md [hunk 4]
```diff
@@ -108,6 +108,15 @@ As long as you can write some function to tell whether the provided point is ins
...
 I can guarantee that we will see similar methods crop up all over the place in the future!
 
+## Video Explanation
+
+Here is a video describing Monte Carlo integration:
+
+<div style="text-align:center">
+<iframe width="560" height="315" src="https://www.youtube.com/embed/AyBNnkYrSWY" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+</div>
+
```
Comment (julianschacher) [inline contents/monte_carlo_integration/monte_carlo_integration.md]: A line too much.
```suggestion
```

File: contents/split-operator_method/split-operator_method.md [hunk 5]
```diff
@@ -161,6 +161,15 @@ As you can see from the code, this involves summing the density, multiplying tha
...
 The Split-Operator method is one of the most commonly used quantum simulation algorithms because of how straightforward it is to code and how quickly you can start really digging into the physics of the simulation results!
 
+## Video Explanation
+
+Here is a video describing the split-operator method:
+
+<div style="text-align:center">
+<iframe width="560" height="315" src="https://www.youtube.com/embed/BBt8EugN03Q" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+</div>
+
```
Comment (julianschacher) [inline contents/split-operator_method/split-operator_method.md]: A line too much.
```suggestion
```

File: contents/stable_marriage_problem/stable_marriage_problem.md [hunk 6]
```diff
@@ -18,6 +18,15 @@ To be clear, even though this algorithm seems conceptually simple, it is rather
...
 I am incredibly interested to see what you guys do and how you implement the algorithm.
 
+## Video Explanation
+
+Here is a video describing the stable marriage problem:
+
+<div style="text-align:center">
+<iframe width="560" height="315" src="https://www.youtube.com/embed/A7xRZQAQU8s" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+</div>
+
```
Comment (julianschacher) [inline contents/stable_marriage_problem/stable_marriage_problem.md]: A line too much.
```suggestion
```

File: contents/tree_traversal/tree_traversal.md [hunk 7]
```diff
@@ -269,6 +269,15 @@ And this is exactly what Breadth-First Search (BFS) does! On top of that, it can
...
 {% endmethod %}
 
+## Video Explanation
+
+Here is a video describing tree traversal:
+
+<div style="text-align:center">
+<iframe width="560" height="315" src="https://www.youtube.com/embed/cZPXfl_tUkA" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+</div>
+
```
Comment (julianschacher) [inline contents/tree_traversal/tree_traversal.md]: A line too much.
```suggestion
```

File: contents/verlet_integration/verlet_integration.md [hunk 8]
```diff
@@ -193,6 +193,15 @@ Unfortunately, this has not yet been implemented in LabVIEW, so here's Julia cod
 
 Even though this method is more widely used than the simple Verlet method mentioned above, it unforunately has an error term of $$\mathcal{O}(\Delta t^2)$$, which is two orders of magnitude worse. That said, if you want to have a simulaton with many objects that depend on one another --- like a gravity simulation --- the Velocity Verlet algorithm is a handy choice; however, you may have to play further tricks to allow everything to scale appropriately. These types of simulatons are sometimes called *n-body* simulations and one such trick is the Barnes-Hut algorithm, which cuts the complexity of n-body simulations from $$\sim \mathcal{O}(n^2)$$ to $$\sim \mathcal{O}(n\log(n))$$.
 
+
```
Comment (julianschacher) [inline contents/verlet_integration/verlet_integration.md]: A line too much.
```suggestion
```

**Pull request merged**